File: forum-backend/src/main/java/com/projects/pes/forumbackend/pojo/Post.java
package com.projects.pes.forumbackend.pojo;

import com.projects.pes.forumbackend.entities.PostEntity;
import com.projects.pes.forumbackend.entities.ResourceEntity;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.OneToMany;

import java.util.List;
import java.util.Set;
import java.util.UUID;

public record Post(
         UUID id,
         String type,
         String title,
         String content,
         UUID posterId,
         String posterName,
         String posterImgUrl,
         UUID parentId,
         List<Post> posts,
         List<Resource> resources
) {
}

File: student-forums-frontend/src/app/post.ts
import { Resource } from "./resource";

export class Post {
    id:string;
    type:string;
    title:string;
    content:string;
    posterId:string;
    posterName:string="";
    posterImgUrl:string="";
    parentId:string|null;
    posts:Array<Post>;
    resourceIds:Array<string> = [];
    constructor(
        id:string,
        type:string,
        title:string,
        content:string,
        posterId:string,
        parentId:string|null,
        posts:Array<Post>,
        resourceIds:Array<string>
    ) {
        this.id = id;
        this.type = type;
        this.title = title;
        this.content = content;
        this.posterId = posterId;
        this.parentId = parentId;
        this.posts = posts;
        

    }
}
